DTML.EduBot/Dialogs/QnaLuisDialog.cs

```diff
@@ -57,7 +58,18 @@ private QnaService MakeQnaServiceFromAttributes()
 
         protected virtual async Task QnaHandler(IDialogContext context, IQnaResult result)
         {
-            await context.PostAsync(result.Answer);
+            await ConvertAndPostResponse(context, result.Answer);
+        }
+
+        protected static async Task ConvertAndPostResponse(IDialogContext context, string result)
+        {
+            string translatedText = await MessageTranslator.TranslateTextAsync(result);
+            if (translatedText == null)
+            {
+                translatedText = $"Sorry I did not get that. Please repeat.";
```

Comment:
Can we instead return the last message as is?

Also, we need to check for empty strings here. The api calls for Null strings are wasteful.